### PR TITLE
Feature/refactor library

### PR DESCRIPTION
**Pull request by glow-mdsol**
Refactored the library layout to reflect changes in std library
Added formatting for XML
Changed Pythonic Variable names to Go

### CODE REVIEW

File: examples/api_gateway_example.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/*
+A Sample API Gateway usage
+*/
+
+
+ func createClient()(mauthClient *go_mauth_client.MAuthClient, err error){
+ 	// load the configuration from the environment
+ 	mauthApp, err := loadApp()
+ 	if err != nil {
+		log.Fatal("Unable: to load client configuration; " +
```
Comment (johnduhart) [inline examples/api_gateway_example.go]: It should return at this point, no?

File: cmd/go_mauth_client/main.go [hunk 2]
```diff
@@ -6,13 +6,14 @@ import (
...
 	"fmt"
-	"github.com/mdsol/go-mauth-client/go_mauth_client"
+	"github.com/mdsol/go-mauth-client"
 	"io/ioutil"
 	"log"
 	"mime"
 	"net/http"
 	"net/url"
 	"os"
+	"github.com/go-xmlfmt/xmlfmt"
```
Comment (johnduhart) [inline cmd/go_mauth_client/main.go]: It would probably be a good idea to sort the imports.

File: examples/api_gateway_example.go (hunk 1 above)
Comment (glow-mdsol) [reply examples/api_gateway_example.go]: `log.Fatal` logs a message and then runs `os.Exit(1)`.  It suicides the application as I understand so returning is not required.

File: cmd/go_mauth_client/main.go (hunk 2 above)
Comment (glow-mdsol) [reply cmd/go_mauth_client/main.go]: Good idea!

**Pull request merged**